src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js

```diff
@@ -68,29 +74,131 @@ class MapViewVisualization extends Visualization {
+     * @param parsedData - All the parsed data to create GeoPoints from.
+     * @param accentColor - accent color of IDE if element doesn't specify one.
+     */
+    prepareDataPoints(parsedData, preparedDataPoints, accentColor) {
+        let latitude  = 0.0;
+        let longitude = 0.0;
+        let zoom      = 11;
+
+        if (parsedData.type === GEO_POINT) {
+            this.pushGeoPoint(preparedDataPoints,parsedData,accentColor);
+            latitude  = parsedData.latitude;
+            longitude = parsedData.longitude;
+        } else if (Array.isArray(parsedData) && parsedData.length && parsedData[0].type === GEO_POINT) {
+            const computed = this.prepareDataPointsHelper(parsedData,preparedDataPoints,accentColor);
+            latitude       = computed.latitude;
+            longitude      = computed.longitude;
+        } else {
+            if (parsedData.type === SCATTERPLOT_LAYER && parsedData.data.length) {
+                const computed = this.prepareDataPointsHelper(parsedData.data,preparedDataPoints,accentColor);
+                latitude       = computed.latitude;
+                longitude      = computed.longitude;
+            } else if (parsedData.type === GEO_MAP && parsedData.layers !== undefined) {
+                parsedData.layers.forEach(layer => {
+                    if (layer.type === SCATTERPLOT_LAYER) {
+                        let dataPoints = layer.data || [];
+                        const computed = this.prepareDataPointsHelper(dataPoints,preparedDataPoints,accentColor);
+                        latitude       = computed.latitude;
+                        longitude      = computed.longitude;
+                    } else {
+                        console.log("Currently unsupported deck.gl layer.")
+                    }
+                })
+            }
+        }
+        return {latitude,longitude,zoom}
+    }
+
+    /**
+     * Helper for prepareDataPoints, calculating also central point.
+     * @returns {{latitude: number, longitude: number}} - center.
+     */
+    prepareDataPointsHelper(dataPoints,preparedDataPoints,accentColor) {
+        let latitudes  = [];
+        let longitudes = [];
+        dataPoints.forEach(e => {
+            this.pushGeoPoint(preparedDataPoints,e,accentColor);
+            latitudes.push(e.latitude);
+            longitudes.push(e.longitude);
+        });
+        let latitude  = 0.0;
+        let longitude = 0.0;
+        if (latitudes.length && longitudes.length) {
+            let minLat = Math.min.apply(null,latitudes);
+            let maxLat = Math.max.apply(null,latitudes);
+            latitude   = (minLat + maxLat)/2;
+            let minLon = Math.min.apply(null,longitudes);
+            let maxLon = Math.max.apply(null,longitudes);
+            longitude  = (minLon + maxLon)/2;
+        }
+        return {latitude,longitude};
+    }
+
+    /**
+     * Pushes a new deck.gl-compatible point from GeoPoint.
```

Comment:
I dont understand the docs. So it pushes things. The docs tell that it pushes a single point, but the argument is a list of "geoPoints" (the name is wrong there). Moreover, what does it mean to push a point FROM GeoPoint? What is "pushing from"? I mean, I really dont get this sentence at all. Moreover, what are the other arguments? If this function only pushes a point, why do I have to provide it with accentColor and "element", which is not defined as well - I only know that "element" is "element" .... "to create new point from", but what is "element" ?

Reply:
Yeah, improved 😄